lib/modules/manager/pep723/schema.ts

```diff
@@ -0,0 +1,18 @@
+import { z } from 'zod';
+import { Toml } from '../../../util/schema-utils';
+import { depTypes, pep508ToPackageDependency } from '../pep621/utils';
+
+const Pep723Dep = z
+  .string()
+  .transform((dep) => pep508ToPackageDependency(depTypes.dependencies, dep));
+
+export const Pep723Schema = Toml.pipe(
+  z
+    .object({
```

Comment:
PEP 723 also supports [setting `requires-python`](https://packaging.python.org/en/latest/specifications/inline-script-metadata/#script-type), and I was wondering if this is necessary to have as a Python constraint. I assumed that this is useful if we actually run a tool to update dependencies in a lock file (like Poetry or uv), but maybe it's also useful without a lock file to update?

Reply:
yes, extract it, the datasource can use it to filter compatible releases